ironfish-rust/src/errors.rs

```diff
@@ -39,48 +48,75 @@ pub enum IronfishError {
     InvalidTransactionVersion,
     InvalidViewingKey,
     InvalidWord,
-    Io(io::Error),
+    Io,
     IsSmallOrder,
     RandomnessError,
-    TryFromInt(num::TryFromIntError),
-    Utf8(string::FromUtf8Error),
+    TryFromInt,
+    Utf8,
     VerificationFailed,
 }
 
+impl IronfishError {
+    pub fn new(kind: IronfishErrorKind) -> Self {
+        Self {
+            kind,
+            source: None,
+            backtrace: Backtrace::capture(),
+        }
+    }
+
+    pub fn new_with_source(kind: IronfishErrorKind, source: Option<Box<dyn Error>>) -> Self {
+        Self {
+            kind,
+            source,
+            backtrace: Backtrace::capture(),
+        }
+    }
```

Comment:
Nit: I would change this to:

```rust
    pub fn new_with_source<E>(kind: IronfishErrorKind, source: E) -> Self 
        where E: Into<Box<dyn Error>>
    {
        Self {
            kind,
            source: Some(source.into()),
            backtrace: Backtrace::capture(),
        }
    }
```

so that instead of writing:

```rust
IronfishError::new_with_source(IronfishErrorKind::TryFromInt, Some(Box::new(e)))
```

you can write:

```rust
IronfishError::new_with_source(IronfishErrorKind::TryFromInt, e)
```

Reply:
Neat! Thanks for the sugar 🍬 🌟